src/backend/expungeservice/parsers/case_parser.py

```diff
@@ -0,0 +1,119 @@
+from html.parser import HTMLParser
+from . import case
+
+
+class CaseParser(HTMLParser):
+    BASE_URI = 'https://publicaccess.courts.oregon.gov/PublicAccessLogin/'
+
+    def __init__(self):
+        HTMLParser.__init__(self)
+        self.cases = []
+        self.column = 0
+        self.within_tr_tag = False
+        self.within_nested_tr = False
+        self.collect_data = False
+        self.case_number = ''      # column 1
+        self.citation_number = []  # column 2
+        self.info = []             # column 3
+        self.date_location = []    # column 4
+        self.type_status = []      # column 5
+        self.charges = []          # column 6+
+        self.case_detail_link = ''
+
+    def handle_starttag(self, tag, attrs):
+        self.__increase_column_count(tag)
+        self.__assign_case_link(tag, attrs)
+        self.__set_flags(tag)
+
+    def handle_endtag(self, tag):
+        if self.__exiting_nested_table(tag):
+            self.within_nested_tr = False
+        elif self.__collect_tr_data(tag):
+            if self.__valid_row():
+                self.__record_case()
+                self.__reset_case()
+            self.__reset_flags()
+
+    def handle_data(self, data):
+        if self.__within_valid_table_row():
+            switcher = {
+                1: self.__column_1,
+                2: self.__column_2,
+                3: self.__column_3,
+                4: self.__column_4,
+                5: self.__column_5,
+            }
+            switcher.get(self.column, self.__column_6_plus)(data)
+
+        elif 'Charge(s)' == data:
+            self.collect_data = True
```

Comment:
It seems this should always be true once we finish parsing the header (<th>) which would occur once per page. So can we just start collecting data for the table rows?

Reply:
yeah that's what its doing. There are five `<th>` tags in the document. `Charge(s)` is the very last `<th>` tag data. Once it passes that point it starts collecting data for the rest of the page.

Reply:
There are actually four separate tables on the page. The table we collect data from is the last table, hence the reason to check for `Charge(s)` table data. Once we pass that point data the data collection is turned on.

Reply:
Ah, I see. I assumed we had only one table on the page. Thanks for the clarification!